File: utils/redis.go
package utils

import (
	"fmt"
	"log"

	"github.com/redis/go-redis/v9"
)

func InitRedis() *redis.Client {
	addr := fmt.Sprintf("%s:%s",
		Load().RedisHost,
		Load().RedisPort,
	)

	client := redis.NewClient(&redis.Options{
		Addr:     addr,
		Password: Load().RedisPassword,
		DB:       0,
	})

	log.Println("Redis connected successfully")
	return client
}

File: src/db/redis.js
const redis = require("redis");

const redisClient = redis.createClient({
  host: process.env.REDIS_HOST,
  port: process.env.REDIS_PORT,
  password: process.env.REDIS_PASSWORD || undefined
});

redisClient.on("error", (err) => {
  console.log("Redis Client Error", err);
});

redisClient.on("connect", () => {
  console.log("Connected to Redis");
});

module.exports = redisClient;